metabase/metabase
# Add public link creation user flow

Epic #35961

### Description

Adds the public link popover for the dashboard and question embed popover. This popover, when opened, should create a public link for the resource and allow users to copy the link. Admins should also see a `Remove public link` button that will allow them to remove the public link to the resource.

### How to verify

There's two scenarios, admin and non-admin. Ensure that public sharing is enabled for the instance, as the public link popover should never appear if public sharing is off.

If the user is an admin:
* They should see the embed popover when they click on the embed button, and see `Create a public link`
* They should also see `Remove public link`, and clicking it should remove the link.
* If the link has already been created, the initial popover should say `Public link` instead of `Create a public link`. Clicking `Public link` should give the same URL as they have seen before.

If the user is a non-admin:
* They should only see the Embed button if a public link has been created.
* They should be able to copy the public link
* They shouldn't be able to see `Remove public link`

### Screenshots

**_An admin should see:_**

_When a link hasn't been created_
<img width="252" alt="image" src="https://github.com/metabase/metabase/assets/25306947/408b2eba-0b6f-42d3-8a47-9f600e9c1298">

_When a link has been created_
<img width="286" alt="image" src="https://github.com/metabase/metabase/assets/25306947/d0577011-aa2c-4747-b9ba-195975b057e7">

<img width="517" alt="image" src="https://github.com/metabase/metabase/assets/25306947/9522ca00-59cb-46f8-bee3-038e0c647471">



_**A non-admin should see:**_

_When a link has been created_
<img width="930" alt="image" src="https://github.com/metabase/metabase/assets/25306947/33077390-14ab-4d12-87f5-4a6e0d179283">

_When no link exists - no embed button_
<img width="166" alt="image" src="https://github.com/metabase/metabase/assets/25306947/99875b2e-0ee0-4555-9bc1-576e424b8a2e">



### Checklist
